workers/unity/Packages/com.improbable.gdk.core/GameObjectRepresentation/GameObjectCreation/SpatialOSEntity.cs

```diff
@@ -0,0 +1,32 @@
+﻿using Improbable.Worker;
+using Unity.Entities;
+
+namespace Improbable.Gdk.Core.GameObjectRepresentation
+{
+    /// <summary>
+    ///     Used to easily retrieve information about a SpatialOS Entity instance from a Unity ECS Entity instance.
+    /// </summary>
+    public struct SpatialOSEntity
```

Comment:
I still don't think this struct is really needed. I understand that it was created to hide the ECS part, however, users still need to know about `IComponentData` to use it and probably create some kind of system to actually create & add it to a monobehaviour. To me it looks like an unnecessary wrapper for fetching the component data storing the actual entity id.

Reply:
We already use the `IComponentData` structs as what the Readers give you, and they just have the fields defined in schema without any unexpected parts anyway. I feel like it's weird to say "you can just use GameObjects and MonoBehaviours for everything" and then have the users do a bunch of `entityManager.GetComponent<>()` in order to create their GameObjects in the first place. Would you suggest just passing the Entity and the EntityManager to the EntityGameObjectCreator user code?